File: intellij-plugin/src/main/kotlin/dev/skillbill/intellij/application/PreferenceCachePort.kt
package dev.skillbill.intellij.application

import dev.skillbill.intellij.domain.DEFAULT_REFRESH_INTERVAL_SECONDS
import dev.skillbill.intellij.domain.LastKnownDisplayCache

/**
 * Lightweight preference and optional last-known display cache port.
 * Persists only settings (CLI executable, refresh interval) and an
 * observation-time-marked display cache. Never writes Skill Bill runtime DBs.
 */
interface PreferenceCachePort {
    fun getCliExecutableOverride(): String?

    fun setCliExecutableOverride(path: String?)

    fun getRefreshIntervalSeconds(): Long

    fun setRefreshIntervalSeconds(seconds: Long)

    fun getLastKnownDisplayCache(): LastKnownDisplayCache?

    fun setLastKnownDisplayCache(cache: LastKnownDisplayCache?)

    companion object {
        fun defaultRefreshIntervalSeconds(): Long = DEFAULT_REFRESH_INTERVAL_SECONDS
    }
}

File: vscode-extension/src/application/PreferenceCachePort.ts
import { DEFAULT_REFRESH_INTERVAL_SECONDS } from "../domain/Constants";
import { LastKnownDisplayCache } from "../domain/LastKnownDisplayCache";

export interface PreferenceCachePort {
  getCliExecutableOverride(): string | undefined;
  setCliExecutableOverride(path: string | undefined): void;
  getRefreshIntervalSeconds(): number;
  setRefreshIntervalSeconds(seconds: number): void;
  getLastKnownDisplayCache(): LastKnownDisplayCache | undefined;
  setLastKnownDisplayCache(cache: LastKnownDisplayCache | undefined): void;
}

export function defaultRefreshIntervalSeconds(): number {
  return DEFAULT_REFRESH_INTERVAL_SECONDS;
}
